src/pages/home/owners/Songs.tsx

```diff
@@ -47,9 +55,27 @@ const Songs: FunctionComponent = () => {
 
   const hlsJsParams = useMemo(
     () => ({
-      onPlaySong: ({ id }: Song) => setCurrentPlayingSongId(id),
-      onStopSong: () => setCurrentPlayingSongId(undefined),
-      onSongEnded: () => setCurrentPlayingSongId(undefined),
+      onPlaySong: ({ id }: Song) => {
+        setPlayerState((prevState) => ({
+          ...prevState,
+          isReadyToPlay: false,
+          currentPlayingSongId: id,
+        }));
+      },
+      onStopSong: () => {
+        setPlayerState((prevState) => ({
+          ...prevState,
+          isReadyToPlay: false,
+          currentPlayingSongId: undefined,
+        }));
+      },
+      onSongEnded: () => {
+        setPlayerState((prevState) => ({
+          ...prevState,
+          isReadyToPlay: false,
+          currentPlayingSongId: undefined,
+        }));
+      },
```

Comment:
Can all of `playerState`/`hlsJsParams` Memo/`useEffects` be refactored into a custom hook and then referenced in each of these components, `PlaySong.tsx`/`InvitesTable.tsx`/`Songs.tsx` or does each component require custom tweaks right now?